### PR TITLE
Name tweaks

### LINKED ISSUE #30: Tweaking names
Some of the main key names we use have become confusing

Annotation objects are stored under a key called labelset.  Collectively these make up a table of annotations and metadata about them. 

    "labelset": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/Annotation"

Each annotation is part of a set of annotations with a particular name.  Membership is indicated using the field (column header in table representation) cellannotation_setname.  Metadata about each set is stored in another table (list of objects) under the key 'cellannotation_setname_metadata'.  These names are ugly, confusing - hard to use in discussion and hard to remember.

Suggested simplification:

labelset --> annotations. # This is a nice, explicit description of content.  Much less confusing than the current name
cellannotation_setname_metadata --> labelsets
cellannotation_setname --> labelset  # This elides whether a labelset refers to a set of annotations united under a single name or a single annotation,  but I think that is OK. 

1. labelsets -> Annotations
2. cellannotation_set --> labelset_name - the name for a set collection of annotations.

### PR DESCRIPTION
**Pull request by dosumis**
Fixes #30 - please see ticket for justification.

@mfutey & @evanbiederstedt  - to review the effect of this PR please just review [examples/Siletti_expression_evidence.json](https://github.com/cellannotation/cell-annotation-schema/pull/31/files#diff-edfbc751c8eb8eb5461175dd349e269bc017bfe75cb146aa2c65ee2405fdac3c) + README.

(This turned out to be more hassle to complete than I expected, but was a useful exercise in testing new infrastructure and editing SOPs.)

### CODE REVIEW

**Comment by dosumis** [comment]
@hkir-dev @ubyndr - schema is good.  Not sure why unit tests are failing.  Surely unit tests should be for code only and so independent of changes to schema and test examples?

**Comment by dosumis** [comment]
Update - partly fixed by updating key BICAN extension.

File: general_schema.json [hunk 1]
```diff
@@ -160,11 +160,11 @@
       "description": "This MUST be a valid ORCID for the author",
       "type": "string"
     },
-    "cellannotation_setname_metadata": {
+    "labelsets": {
       "type": "array",
       "$ref": "#/definitions/cellannotation_setname_metadata"
```
Comment (ubyndr) [inline general_schema.json:165]: Is this part correct? There is no cellannotation_setname_metadata section in the definitions

Comment (ubyndr) [inline general_schema.json:165]:
```suggestion
      "$ref": "#/definitions/labelsets"
```

**Comment by ubyndr** [comment]
https://github.com/cellannotation/cell-annotation-schema/blob/655051be62f9a49d6e8e8ed88a35005368be83da/CAP_extension.json#L8-L13
This part is missing labelsets cc @dosumis

File: general_schema.json (hunk 1 above)
Comment (dosumis) [reply general_schema.json]: Could have sworn I'd already fixed that.  Thanks for spotting.  Wondering why it didn't show up on the JSON schema validation side of the checks.

File: src/test/test_data/CAP_extension.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        "$ref": "./general_schema.json"
+      }
+    ],
+
+  "required": [
+    "cellannotation_schema_version",
+    "cellannotation_timestamp",
+    "cellannotation_version",
+    "cellannotation_url"
+  ]
```
Comment (ubyndr) [inline src/test/test_data/CAP_extension.json:13]:
```suggestion
  "required": [
    "cellannotation_schema_version",
    "cellannotation_timestamp",
    "cellannotation_version",
    "cellannotation_url",
    "labelsets"
  ]
```
cc @hkir-dev

**Comment by dosumis** [comment]
I think this is too strict for a test. The schema should do the work.

image

I will fix for now to ensure PR passes.

**Comment by hkir-dev** [comment]
I think there is a bug in the merge logic. I am trying to fix it now. You don't need to change your schema.

**Comment by dosumis** [comment]
> I think there is a bug in the merge logic. I am trying to fix it now. You don't need to change your schema.

I looks to me like the problem was with overly strict the tests - see #34

**Comment by hkir-dev** [comment]
Unit tests should only test the behaviour of the code. Updating the schema shouldn't fail unit tests. Code should work with any schema. So I am trying to test the code with the schemas located in the `test_data` folder. 

However, there is a bug in the schema merging, so `test_data/*_extension.json` schemas are accidentally merged with the `./general_schema.json` instead of the `test_data/general_schema.json`.

**Review by ubyndr — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
-# cell-annotation-schema
+# cell-annotation-schema (CAS)
 
 General, open-standard schema for cell annotations
 
 ## Motivation
 
-Annotation of single cell transcriptomics data with cell types/classes is inherently variable. The reasons that authors choose to annotate some particular set of cells in a particular way is not typically represented in annotated data and there is no established standard for doing so - at best it is accessible by reading the associated paper, but as datasets information in the paper on evidence is often incomplete and hard to resonstruct.  
+Annotation of single cell transcriptomics data with cell types/classes is inherently variable. The reasons that authors choose to annotate a set of cells with a particular label are not typically represented in annotated data and there is no established standard for doing so.  For relatively simple datasets it may be possible to reconstruct this information by reading an associated publicaion, but as single cell transcriptomics datasets and accompanying publications become increasingly complex, it is increaingly impossilbe to do so.
```
Comment (mfutey) [inline README.md:8]: "by reading an associated publication" (publication is missing a "t")

**Review by mfutey — APPROVED** [review]

**Review by evanbiederstedt — APPROVED** [review]

**Comment by evanbiederstedt** [comment]
Let's run spellcheck on all of this please

**Comment by evanbiederstedt** [comment]
@dosumis We really should run this through spellcheck

**Pull request merged**